src/v/cloud_roles/apply_abs_oauth_credentials.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_roles/apply_credentials.h"
+
+namespace cloud_roles {
+
+/// Azure Managed Identities uses Bearer tokens to authorize a request
+class apply_abs_oauth_credentials final : public apply_credentials::impl {
```

Comment:
Please add a descriptive comment to this class. My guess at what the comment should be is something like

```cpp
/*
* apply_abs_oauth_credentials is an implementation of apply_credentials that
*  applies the Bearer header for authorize requests in support of Azure
* Managed Identities.
*/
```